Dockerfile

```diff
@@ -2,7 +2,7 @@ FROM crystallang/crystal:latest-alpine AS builder
 RUN apk update && apk upgrade && apk add sqlite-static
 WORKDIR /build/
 ARG version
-RUN git clone --branch ${version:-v2.3.0} --depth 1 https://github.com/toddsundsted/ktistec .
+RUN git clone --branch ${version:-main} --depth 1 https://github.com/toddsundsted/ktistec .
```

Comment:
@JayVii i merged this, however after merging it occurred to me that this may be better set to the `dist` branch since tags will be applied to commits there and not on `main`. generally, i _probably have merged_ `main` into `dist`, so it's probably okay, but it could pick up some unintentional commits.

Reply:
My thinking was, that during development (if using a docker-based workflow...) it probably makes sense to build from main by default. Using the `--build-arg version=some-commit-or-tag` argument to docker, one can always build from a specific branch, tag or commit.

In fact, that is exactly what the pipeline does too (see: https://github.com/toddsundsted/ktistec/blob/1490b81c345388723a987740bb4c86b82b14616a/.github/workflows/cd.yml#L57). It sets the `version` variable to the current tag (which is on the dist branch, as you said) and builds from there.

Only if that argument was not given to docker, it tries to build from the default (main).

Possibly, my idea was wrong and it makes more sense to set `dist` as default, you ae the expert :) . As I am not developing ktistec or am typically not running builds from non-tags, I am not really sure what makes sense here.

Reply:
okay, this makes sense! thanks for clarifying!